staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -227,29 +231,31 @@ type ConditionFunc func(info *resource.Info, o *WaitOptions) (finalObject runtim
 // RunWait runs the waiting logic
 func (o *WaitOptions) RunWait() error {
 	visitCount := 0
-	err := o.ResourceFinder.Do().Visit(func(info *resource.Info, err error) error {
-		if err != nil {
+	for {
+		err := o.ResourceFinder.Do().Visit(func(info *resource.Info, err error) error {
+			if err != nil {
+				return err
+			}
+			visitCount++
+			finalObject, success, err := o.ConditionFn(info, o)
+			if success {
+				o.Printer.PrintObj(finalObject, o.Out)
+				return nil
+			}
+			if err == nil {
+				return fmt.Errorf("%v unsatisified for unknown reason", finalObject)
+			}
 			return err
+		})
+		if apierrors.IsNotFound(err) && o.NonExistent {
+			time.Sleep(1 * time.Second)
```

Comment:
default time out is 30 seconds. you can use o.Timeout here

Reply:
@hpandeycodeit I've been thinking about this, maybe hardcoding a wait of 5s instead of 1. This is the time between the loops in the for block, so I'll wait 1s between each get. 

The default timeout of 30s here for me doesn't apply, as what's going to happen is that after 30s it will timeout not going through the loop again